lib/wmic.js

```diff
@@ -1,106 +1,118 @@
+    //
+    // CreationDate               KernelModeTime  ParentProcessId  ProcessId  UserModeTime  WorkingSetSize
+    // 20150329221650.080654+060  153750000       0                777        8556250000    110821376
 
     stdout = stdout.split(os.EOL)
 
+    var again = false
     var statistics = {}
     for (var i = 1; i < stdout.length; i++) {
-      var line = stdout[i].replace(/\s\s+/g, ' ').split(' ')
-
-      // results are in alphabetical order
-      var id = parseInt(line[1], 10)
-      var hst = options.history[id]
-      var workingsetsize = parseFloat(line[3])
-      var stats = {
-        kernelmodetime: parseFloat(line[0]),
-        usermodetime: parseFloat(line[2])
-      }
+      var line = stdout[i].trim().replace(/\s\s+/g, ' ').split(' ')
 
-      // process usage since last call (obscure sheeeeet don't ask)
-      var total = (stats.kernelmodetime - (hst.kernelmodetime || 0) + stats.usermodetime - (hst.usermodetime || 0)) / 10000000
-      // time elapsed between calls
-      var seconds = hst.uptime !== undefined ? uptime - hst.uptime : 0
-      var cpu = seconds > 0 ? (total / seconds) * 100 : 0
-
-      options.history[id] = stats
-      options.history[id].uptime = uptime
-
-      statistics[id] = {
-        cpu: cpu,
-        memory: workingsetsize,
-        time: stats.usermodetime + stats.kernelmodetime, // should be ms according to https://msdn.microsoft.com/en-us/library/aa394372(v=vs.85).aspx
-        start: new Date(date - (stats.usermodetime + stats.kernelmodetime)),
-        pid: id
+      if (!line || line.length !== 6) {
+        continue
       }
-    }
 
-    var results = []
-    for (i = 0; i < pid.length; i++) {
-      if (statistics[pid[i]]) {
-        results[i] = statistics[pid[i]]
-        continue
+      var creation = parseDate(line[0])
+      var ppid = parseInt(line[2], 10)
+      var pid = parseInt(line[3], 10)
+      var kerneltime = Math.round(parseInt(line[1], 10) / 10000)
+      var usertime = Math.round(parseInt(line[4], 10) / 10000)
+      var memory = parseInt(line[5], 10)
+
+      var hst = history.get(pid)
+      if (hst === undefined) {
+        again = true
+        hst = {ctime: kerneltime + usertime, uptime: uptime}
       }
 
-      results[i] = {cpu: 0, memory: 0, pid: parseInt(pid[i], 10), time: 0, start: new Date(date)}
+      // process usage since last call
+      var total = (kerneltime + usertime - hst.ctime)
+      // time elapsed between calls
+      var seconds = uptime - hst.uptime
+      var cpu = seconds > 0 ? (total / (seconds * 1000)) * 100 : 0
+
+      history.set(pid, {ctime: usertime + kerneltime, uptime: uptime})
+
+      statistics[pid] = {
+        cpu: Math.min(Math.round(cpu * 1000) / 1000, 100.0),
+        memory: memory,
+        ppid: ppid,
+        pid: pid,
+        ctime: usertime + kerneltime,
+        elapsed: date - creation.getTime(),
+        timestamp: date
+      }
     }
 
-    done(null, results)
+    if (again) {
```

Comment:
This I kinda disagree because it makes the first call really slow no? I agree that the data will be accurate only from the second call on windows but we should just state so in the README.

Reply:
If we don't do this the readings are just wrong.
And yes only the first call takes longer.

> I agree that the data will be accurate only from the second call

That's not true. Actually in the previous version you use the history also if that history was really old (like 10 hours ago) instead now the previous values are used only if they are not older than a minute. 

This means that:
- if you call pidusage frenquently you get "fast" and "correct" readings.
- if you call pidusage sporadically you get "correct" readings.